### DISCUSSION TITLE
High RAM usage (20MB RSS of each mcjoin process)

### DISCUSSION THREAD

**Discussion by jcpvdm** [troglobit/mcjoin#32] [General]
A single mcjoin process running just as `mcjoin 232.4.4.4` consumes immediately ~20MB of RAM (RSS). 
It seems way too high... do you have an idea how to reduce it?

**Comment by troglobit** [comment]
"seems way too high", really? Based on what, compared to what? What is *not* too high according to you?

I've made my tool free for you to download and use, regardless if it's for personal or corporate use, and still you're unhappy with the performance?!

Did you know I've made other tools? Tools that have less features, like not the impressive graphics, not the support for multiple address families, tools that cannot run on any UNIX system, but only send a single ioctl to the Linux kernel to join a single ASM group. Those tools likely use A LOT less RSS … but that wouldn't make you happy either. So maybe I should rewrite the tool in assembler for your favourite SoC, fit it in less than a single page of RAM. No, still not enough? 

It's all very simple. If you don't like it, write your own tool then. Even better, I've licensed the tool so that YOU can fork it and improve upon it. Amazing right?!

**Reply by jcpvdm** [reply]
First of all I think this tool is amazing and thank you for it! This is not a complaint post.

While testing multicast with many mcjoin receiver processes, noticed that system available memory quickly depleted.
I was surprised to see 20MB RSS for each process, since in my mind the task was simple (join a single group, count packets, display text based output in terminal or log). So I just wondered what is contributing to 20MB and if it could be easily improved.

I know you have other multicast tools (smcroute, mreceive) and will probably try them out for scaled setup.

**Reply by troglobit** [reply]
Fixed in 2fd1916ff0e3fa9e7d037ebb309bb8138945a7f7
